react/features/app/actions.js

```diff
@@ -31,176 +31,72 @@ declare var APP: Object;
  * @returns {Function}
  */
 export function appNavigate(uri: ?string) {
-    return (dispatch: Dispatch<any>, getState: Function) =>
-        _appNavigateToOptionalLocation(dispatch, getState, parseURIString(uri));
-}
-
-/**
- * Triggers an in-app navigation to a specific location URI.
- *
- * @param {Dispatch} dispatch - The redux {@code dispatch} function.
- * @param {Function} getState - The redux function that gets/retrieves the redux
- * state.
- * @param {Object} newLocation - The location URI to navigate to. The value
- * cannot be undefined and is assumed to have all properties such as
- * {@code host}, {@code contextRoot}, and {@code room} defined. Depending on the
- * property, it may have a value equal to {@code undefined} and that may be
- * acceptable.
- * @private
- * @returns {Promise<void>}
- */
-function _appNavigateToMandatoryLocation(
```

Comment:
Could you provide an explanation for why you de-encapsulated parts of the config load logic?

In the past I had trouble understanding this logic and specifically the optional/mandatory terminology; I didn't have context and like an amateur didn't ask for it. I think because of that missing context this might be the rare case where I prefer having one bigger function like you did, so that my mental context stays within the same function and its scope.

Reply:
Yes, the whole point was (wishful) readability. Removing things like functions with a single line invoking another function, and the implicitly introduced optional/mandatory navigation (which is a bad naming anyway).